### DISCUSSION TITLE
Why Instance#setBlock doesn't call BlockPlacementRule#onPlace

### DISCUSSION THREAD

**Discussion by Ynverxe** [Minestom/Minestom#2011] [Help]
Currently fluid-simulation of the Vanilla implementation seems broken, since the updating of the fluids is done through a block placement rule using the onPlace and Instance#setBlock method to place the blocks that represent the adjacent fluids, it seems that in its At the moment it was expected that this last method would also call the BlockPlacementRule#onPlace method, that's what I think. So in order to fix this, possible solutions come to mind:
1) That it is a bug and Instance#setBlock should really call BlockPlacementRule#onPlace.
2) If it is not a bug, then I should use the Instance#placeBlock method (I don't know why it is a method marked as internal, please someone explain it to me).
3) Rework the fluid simulation module in another way.

**Comment by KrystilizeNevaDies** [comment]
@Ynverxe Please reopen this issue on the vanilla reimplementation repo

**Reply by Ynverxe** [reply]
Done, https://github.com/Minestom/VanillaReimplementation/issues/50
